Sources/Tracking/CustomerIO.swift

```diff
@@ -153,7 +92,24 @@ public class CustomerIO: CustomerIOInstance {
         region: Region,
         configure configureHandler: ((inout SdkConfig) -> Void)?
     ) {
-        var newSdkConfig = SdkConfig.Factory.create(region: region)
+        initialize(siteId: siteId, apiKey: apiKey, region: region, configParams: [:], configure: configureHandler)
+    }
+
+    /**
+     Initialize the shared `instance` of `CustomerIO` with optional `configParams` to intialize SdkConfig.
+     Call this function when your app launches, before using `CustomerIO.instance`.
+
+     Currently this constructor is of internal use only.
+     */
+    @available(iOSApplicationExtension, unavailable)
+    public static func initialize(
+        siteId: String,
+        apiKey: String,
+        region: Region,
+        configParams: [String: Any] = [:],
+        configure configureHandler: ((inout SdkConfig) -> Void)?
```

Comment:
I don't have any better solution than that, I just wanted to take away the responsibility from wrapper clients to call the modify, as it might create confusion about when to modify it, i.e, after making the changes to `configureHandler` or which changes needs to go first. 

The order might be confusing, that's why I wanted to take away this confusion. 

But have both paramterts: `configParams`, `configure` might be confusing too, because Aman asked the same thing about whats the difference, so maybe,  `SdkConfig.modify()` might be a better call.

Reply:
When I suggest the SDK wrappers contain code similar to:
```
// Code that could exist in the SDK wrappers: 
CustomerIO.initialize(siteId, apiKey, region} { config in 
  config.modify(["foo": "bar"]) 
}
```

I think it would be best if this code wasn't called directly from a customer in their app. I suggest that we provide a nice API to customers to use then the SDK wrapper internally calls this code. 

So, here might be an example piece of code for the RN SDK:

```js
// Here is code that a customer would have in their app. 
// This is a convenient javascript API that javascript developers would be familiar with. 
let config = new CIOConfig()
config.backgroundQueueExpiredSeconds = 0.5

CustomerIO.initialize(siteId: "", apiKey: "", region: "US", config)

// Here is code then that is internal in the RN SDK:
initialize(siteId, apikey, region, configObject) {
  // Note: this is not valid javascript code, but you get the example I am trying to show
  CustomerIO.initialize(siteId, apiKey, region} { config in 
     config.modify(configObject) 
  }
}
```

Would this solution work?

Reply:
This portion is used by our iOS native binding, Customer isn't going to see them, they send `config` once. 

But i have changed the structure.